packages/next/src/build/next-config-ts/utils.ts

```diff
@@ -0,0 +1,71 @@
+import type { Options as SWCOptions } from '@swc/core'
+import type { CompilerOptions } from 'typescript'
+
+import { resolve } from 'node:path'
+import { parseJsonFile } from '../load-jsconfig'
+
+export function resolveSWCOptions(cwd: string): SWCOptions {
+  const tsConfig = lazilyGetTSConfig(cwd)
+
+  return {
+    jsc: {
+      parser: {
+        syntax: 'typescript',
+      },
+      paths: tsConfig?.compilerOptions?.paths,
+      // SWC requires `baseUrl` to be passed when `paths` are used.
+      // Also, `baseUrl` must be absolute.
+      baseUrl: resolve(cwd, tsConfig?.compilerOptions?.baseUrl ?? ''),
+      experimental: {
+        keepImportAttributes: true,
+        emitAssertForImportAttributes: true,
+      },
+    },
+    env: {
+      targets: {
+        node: process.versions.node,
```

Comment:
Add a TODO mentioning that we're not considering other runtimes. No need to fix this short term though.

Reply:
Added!